src/main/js/pullRequest.ts

```diff
@@ -22,13 +22,117 @@
  * SOFTWARE.
  */
 
-import { BasicComment, BasicPullRequest, MergeCommit, PossibleTransition, PullRequest } from "./types/PullRequest";
+import {
+  BasicComment,
+  BasicPullRequest,
+  Comment,
+  Comments,
+  Conflicts,
+  MergeCommit,
+  PossibleTransition,
+  PullRequest,
+  PullRequestCollection
+} from "./types/PullRequest";
 import { apiClient } from "@scm-manager/ui-components";
-import { Branch, Link, Repository } from "@scm-manager/ui-types";
+import { Branch, Changeset, HalRepresentation, Link, Links, PagedCollection, Repository } from "@scm-manager/ui-types";
+import { useMutation, useQuery, useQueryClient } from "react-query";
 
 const CONTENT_TYPE_PULLREQUEST = "application/vnd.scmm-pullRequest+json;v=2";
 
-export function createPullRequest(url: string, pullRequest: BasicPullRequest) {
+// React-Query Hooks
+
+export const prQueryKey = (repository: Repository, pullRequestId: string) => {
+  return ["repository", repository.namespace, repository.name, "pull-request", pullRequestId];
+};
+
+export const usePullRequest = (repository: Repository, pullRequestId: string) => {
+  const { error, isLoading, data } = useQuery<PullRequest, Error>(prQueryKey(repository, pullRequestId), () =>
+    getPullRequest((repository._links.pullRequest as Link).href + "/" + pullRequestId)
+  );
+
+  return {
+    error,
+    isLoading,
+    data
+  };
+};
+
+function getPullRequest(url: string): Promise<PullRequest> {
+  return apiClient.get(url).then(response => response.json());
+}
+
+export const useUpdatePullRequest = (repository: Repository, pullRequest: PullRequest, callback?: () => void) => {
+  const queryClient = useQueryClient();
+  const { isLoading, error, mutate } = useMutation<unknown, Error, PullRequest>(
+    pr => updatePullRequest(pr).then(callback),
```

Comment:
I'm not sure if this is the right place to call the callback. I think `onSuccess` would be safer or just return `isSuccess` and make a declarative redirect from Component.